scene/gui/text_edit.cpp

```diff
@@ -1941,6 +1922,7 @@ void TextEdit::_notification(int p_what) {
 			}
 
 			if (has_focus()) {
+				theme_cache.style_focus->draw(ci, Rect2(Point2(), get_size()));
```

Comment:
I realized that the focus drawing should be before the text drawing, since it might not just be an outline and then it would block the text. LineEdit also has it before the text.